server/interview/interview.service.js

```diff
@@ -1,48 +1,160 @@
-    interview.difficulty = difficulty;
+  let interview = new Interview();
+  interview.owner = user._id;
+  interview.difficulty = difficulty;
 
-    let excludedQuestionIDs = user.questions;
-    let question = await questionService.findARandomQuestionByDifficulty(difficulty, excludedQuestionIDs);
+  let excludedQuestionIDs = user.questions;
+  let question = await questionService.findARandomQuestionByDifficulty(
+    difficulty,
+    excludedQuestionIDs
+  );
+  interview.participants.push({ user: user._id, question: question._id });
+  user.interviews.push(interview._id);
+  user.questions.push(question._id);
 
-    interview.participants.push({ user: user._id, question: question._id });
-    user.interviews.push(interview._id);
-    user.questions.push(question._id);
-    
-    [interview, user] = await Promise.all([interview.save(), user.save()]);
+  [interview, user] = await Promise.all([interview.save(), user.save()]);
 
-    return interview;
+  return interview;
 }
 
+/**
+ * Returns the interview added with the new participant and a new question.
+ *
+ */
 async function addParticipantToAnInterview(user, interviewId) {
-    let interview = await Interview.findById(interviewId);
-    let difficulty = interview.difficulty;
-    
-    let excludedQuestionIDs = user.questions.concat(interview.participants[0].question);
-    //Removing duplicates from array
-    excludedQuestionIDs = excludedQuestionIDs.filter((item, index) => excludedQuestionIDs.indexOf(item) === index);
-    let question = await questionService.findARandomQuestionByDifficulty(difficulty, excludedQuestionIDs).catch((err) => {
-        throw err;
+  let interview = await Interview.findById(interviewId);
+  const participants = interview.participants;
+  if (participants.length >= 2)
+    throw Error("There is already two participants for this interview");
+  let difficulty = interview.difficulty;
+
+  let excludedQuestionIDs = user.questions.concat(
+    interview.participants[0].question
+  );
+  //Removing duplicates from array
+  excludedQuestionIDs = excludedQuestionIDs.filter(
+    (item, index) => excludedQuestionIDs.indexOf(item) === index
+  );
+  let question = await questionService
+    .findARandomQuestionByDifficulty(difficulty, excludedQuestionIDs)
+    .catch((err) => {
+      throw err;
     });
 
-    interview.participants.push({ user: user._id, question: question._id });
-    user.interviews.push(interview._id);
-    user.questions.push(question._id);
-    
-    [interview, user] = await Promise.all([interview.save(), user.save()]);
+  interview.participants.push({ user: user._id, question: question._id });
+  user.interviews.push(interview._id);
+  user.questions.push(question._id);
+
+  [interview, user] = await Promise.all([interview.save(), user.save()]);
+
+  return interview;
+}
+
+/**
+ * Returns all the relevant details of completed interviews of a user.
+ *
+ */
+async function getAllCompletedInterviewsOfAUser(user) {
```

Comment:
Right now you are making many API calls to the database to get the completed interviews 

It's more efficient to do it with one MongoDB query that gets all interviews for that user and where endTime exists 

This can also be applied for getting current interviews as well

Take a look at the different type of queries in MongoDB - https://docs.mongodb.com/manual/tutorial/query-documents/

Reply:
Combined it into a single query